src/core/Akka.Streams.Tests/Dsl/QueueSinkSpec.cs

```diff
@@ -170,10 +170,8 @@ public void QueueSink_should_fail_pull_future_when_stream_is_completed()
                 var result = queue.PullAsync().Result;
                 result.Should().Be(Option<int>.None);
 
-                ((Task)queue.PullAsync()).ContinueWith(t =>
-                {
-                    t.Exception.InnerException.Should().BeOfType<IllegalStateException>();
-                }, TaskContinuationOptions.OnlyOnFaulted).Wait();
+                var exception = Record.ExceptionAsync(async () => await queue.PullAsync()).Result;
+                exception.Should().BeOfType<StreamDetachedException>();
```

Comment:
Using XUnit API to catch the exception and unwrap. Also, this should always run, not just on `OnlyOnFaulted`, to make sure we catch regressions.